packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -98,18 +98,19 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
     status.stop();
   }
 
-  final String? binaryName = getCmakeExecutableName(windowsProject);
-  final File appFile = buildDirectory
+  final Directory outputDirectory = buildDirectory
     .childDirectory('runner')
-    .childDirectory(sentenceCase(buildModeName))
-    .childFile('$binaryName.exe');
+    .childDirectory(sentenceCase(buildModeName));
+  final File appFile = outputDirectory.childFile('${getCmakeExecutableName(windowsProject)}.exe');
   if (appFile.existsSync()) {
-    final String appSize = (buildInfo.mode == BuildMode.debug)
+    final String? directorySize = await getDirectorySize(outputDirectory);
+    final String outputSize = (buildInfo.mode == BuildMode.debug || directorySize == null)
         ? '' // Don't display the size when building a debug variant.
-        : ' (${getSizeAsMB(appFile.lengthSync())})';
-    globals.logger.printStatus(
-      '${globals.logger.terminal.successMark}  '
-      'Built ${globals.fs.path.relative(appFile.path)}$appSize.',
+        : ' ($directorySize)';
+
+    globals.printStatus(
+      '${globals.terminal.successMark} '
+      'Built ${globals.fs.path.relative(outputDirectory.path)}$outputSize.',
```

Comment:
Personally I'd rather print individual file sizes instead of the output directory's size as not all files are necessarily important for the final app size. For example, Windows symbols are large and can be excluded from the app.

Reply:
Agreed, I actually don't think we should measuring directories at all (unless the directory is "package" that will be distributed to end users, as I'm assuming it works on iOS and macOS. Consider this issue on web, where we intentionally create directories that the user web app may never use: I'm actually not sure that we should be showing the entire directory size every time. Consider, for example, [this bug](https://github.com/flutter/flutter/issues/126738) where we're generating contents in the

Reply:
We could avoid showing the directory size for web. Windows is tricky, if I recall, because the `.exe` file is tiny, and the actual app is in an `.so` file. I believe Linux is similar.

Reply:
Opted not to output sizes for Windows and Linux

Reply:
That's a clean compromise, nice 👍